app/uk/gov/hmrc/partnershipidentificationfrontend/controllers/CaptureSautrController.scala

```diff
@@ -55,20 +64,25 @@ class CaptureSautrController @Inject()(mcc: MessagesControllerComponents,
     implicit request =>
       authorised().retrieve(internalId) {
         case Some(authInternalId) =>
-          journeyService.getJourneyConfig(journeyId, authInternalId).flatMap {
-            journeyConfig =>
-              CaptureSautrForm.form.bindFromRequest().fold(
-                formWithErrors => {
-                  Future.successful(
-                    BadRequest(view(journeyId, journeyConfig.pageConfig, routes.CaptureSautrController.submit(journeyId), formWithErrors))
-                  )
-                },
-                sautr =>
-                  partnershipIdentificationService.storeSautr(journeyId, sautr).map {
-                    _ => Redirect(routes.CapturePostCodeController.show(journeyId))
+          CaptureSautrForm.form.bindFromRequest().fold(
+            formWithErrors => {
+              journeyService.getJourneyConfig(journeyId, authInternalId).map {
+                journeyConfig =>
+                  journeyConfig.partnershipType match {
+                    case GeneralPartnership | ScottishPartnership =>
+                      BadRequest(sautr(journeyId, journeyConfig.pageConfig, routes.CaptureSautrController.submit(journeyId), formWithErrors, displaySkipSautrLink = true))
+                    case ScottishLimitedPartnership | LimitedPartnership | LimitedLiabilityPartnership =>
+                      BadRequest(sautr(journeyId, journeyConfig.pageConfig, routes.CaptureSautrController.submit(journeyId), formWithErrors, displaySkipSautrLink = false))
+                    case invalidType =>
+                      throw new InternalServerException(s"Invalid Partnership Type: $invalidType on SAUTR page")
                   }
-              )
-          }
+              }
+            },
+            sautr =>
```

Comment:
Having the view named as sautr and this parameter as well seems a little confusing. Could we change one of them please?